chain/client/src/view_client.rs

```diff
@@ -799,6 +799,10 @@ impl Handler<NetworkViewClientMessages> for ViewClientActor {
                         };
                         ShardStateSyncResponse { header, part: None }
                     }
+                    Ok(false) => {
+                        // Can't pass validation, possible malicious behavior
+                        return NetworkViewClientResponses::NoResponse;
+                    }
                     Err(_) => {
                         // This case may appear in case of latency in epoch switching.
                         // Request sender is ready to sync but we still didn't get the block.
```

Comment:
The logging on the next line is wrong now. The error is not necessarily "cannot get block".
(The logging was already wrong because it was not only matching DbNotFound).
Should be something more generic, like "Failed to verify sync hash valitity"

Reply:
++